src/HtmlParser/Util.elm

```diff
@@ -390,3 +392,279 @@ toVirtualDomEach node =
 toAttribute : (String, String) -> Attribute msg
 toAttribute (name, value) =
   attribute name value
+
+{-| Converts nodes to virtual dom SVG nodes
+-}
+toVirtualDomSvg : List HtmlParser.Node -> List (Svg msg)
+toVirtualDomSvg nodes =
+  List.map toVirtualDomSvgEach nodes
+
+toVirtualDomSvgEach : HtmlParser.Node -> Svg msg
+toVirtualDomSvgEach node =
+  case node of
+    HtmlParser.Element name attrs children ->
+      Svg.node name (List.filterMap toSvgAttribute attrs) (toVirtualDomSvg children)
+
+    HtmlParser.Text s ->
+      text s
+
+    HtmlParser.Comment _ ->
+      text ""
+
+toSvgAttribute : (String, String) -> Maybe (Attribute msg)
+toSvgAttribute (name, value) =
+  case name of
```

Comment:
Yeah, it definitely is tough because you have to have the `_` there. I'll have to think about it. Having all the `Just` there is troublesome, but probably not the end of the world. I might have an idea, but will have to think on it.
This is really cool, though. Great job.